lib/lightning/credentials/credential.ex

```diff
@@ -50,33 +51,45 @@ defmodule Lightning.Credentials.Credential do
       message: "you have another credential with the same name"
     )
     |> assoc_constraint(:user)
-    |> assoc_constraint(:oauth_client)
     |> validate_format(:name, ~r/^[a-zA-Z0-9_\- ]*$/,
       message: "credential name has invalid format"
     )
-    |> validate_oauth()
+    |> maybe_validate_oauth_fields(attrs)
   end
 
-  defp validate_oauth(changeset) do
-    if get_field(changeset, :schema) == "oauth" do
-      body = get_field(changeset, :body) || %{}
+  defp maybe_validate_oauth_fields(changeset, attrs) do
+    is_oauth = get_field(changeset, :schema) == "oauth"
+    oauth_token_id = get_field(changeset, :oauth_token_id)
```

Comment:
This is used only the condition to see if it's not set. I believe thanks to the pairing I can understand better this as where part of the magic happens. For future maintainance, maybe we should give here a domain variable name to the condition. Something related to the purpose like `reusing_token? = get_field(changeset, :oauth_token_id)`?

Reply:
Thanks, that is done